src/relcast.erl

```diff
@@ -463,9 +488,27 @@ handle_actions([{multicast, Message}|Tail], Batch, State =
         {ModuleState, Actions} ->
             handle_actions(Actions++Tail, Batch, update_next(IDs -- [ID], CF, Key, State#state{modulestate=ModuleState, key_count=KeyCount+1}))
     end;
+handle_actions([{callback, Message}|Tail], Batch, State =
+               #state{key_count=KeyCount, bitfieldsize=BitfieldSize, id=ID, ids=IDs, active_cf=CF, module=Module}) ->
+    Bitfield = make_bitfield(BitfieldSize, IDs, ID),
+    Key = make_outbound_key(KeyCount),
+    rocksdb:batch_put(Batch, CF, Key, <<2:2/integer, Bitfield:BitfieldSize/bits, Message/binary>>),
+    case Module:callback_message(ID, Message, State#state.modulestate) of
+        none ->
+            handle_actions(Tail, Batch, update_next(IDs -- [ID], CF, Key, State#state{key_count=KeyCount+1}));
+        OurMessage when is_binary(OurMessage) ->
+            %% handle our own copy of the message
+            %% deferring your own message is an error
+            case Module:handle_message(OurMessage, ID, State#state.modulestate) of
+                ignore ->
+                    handle_actions(Tail, Batch, update_next(IDs -- [ID], CF, Key, State#state{key_count=KeyCount+1}));
+                {ModuleState, Actions} ->
+                    handle_actions(Actions++Tail, Batch, update_next(IDs -- [ID], CF, Key, State#state{modulestate=ModuleState, key_count=KeyCount+1}))
+            end
```

Comment:
Is a fallback case needed here?

Reply:
no, we should crash if the callback module returns anything else.